#166214085 integrate coveralls code climate

Nennyfills opened the pull request:
### Description
Integrate coveralls and codeclimate code coverage service

#### Type of change
- Install coveralls
- Add coveralls badge to readme.md file
- Add codeclimate badge to readme.md file
- Add coveralls-coveraget script
- Add coverage script

#### How Can This Been Tested?
N/A

#### Checklist:
N/A

#### Pivotal story
[#166214085](https://www.pivotaltracker.com/story/show/166214085)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/45515619/58683213-7d64ad00-836b-11e9-8ad5-4ac6b53d5a79.png)

Lundii reviewed (changes requested):
I tested it out. 

* first, you have 4 commits
* your pull request title does not follow the convention
* I ran `npm run coverage` after everything ran, I got the error below. Why is that 
![image](https://user-images.githubusercontent.com/37886814/58649316-b36e4680-8303-11e9-8e49-fc5cd206baa5.png)

Nennyfills commented:
> I tested it out.
> 
> * first, you have 4 commits
> * your pull request title does not follow the convention
> * I ran `npm run coverage` after everything ran, I got the error below. Why is that
>   ![image](https://user-images.githubusercontent.com/37886814/58649316-b36e4680-8303-11e9-8e49-fc5cd206baa5.png)

t

> I tested it out.
> 
> * first, you have 4 commits
> * your pull request title does not follow the convention
> * I ran `npm run coverage` after everything ran, I got the error below. Why is that
>   ![image](https://user-images.githubusercontent.com/37886814/58649316-b36e4680-8303-11e9-8e49-fc5cd206baa5.png)
have fixed the commits
have corrected the PR title
for this to run I need a Travis.yml file, someone else is working on Travis branch.

Adekoreday reviewed (changes requested):
kindly rebase this branch so we can proceed

Lundii reviewed (commented):
The travis branch have been merged. Please do a rebase

allebd reviewed (changes requested):
Awaiting a rebase

Nennyfills commented:
> kindly rebase this branch so we can proceed

done

Nennyfills commented:
> The travis branch have been merged. Please do a rebase

done

Lundii commented:
> > The travis branch have been merged. Please do a rebase
> 
> done

i checked the migration folder of the develop branch (inside database folder). It contains some files your branch doesn't have. Can you please check it out to confirm, because a PR was merged recently.

Adekoreday reviewed (changes requested):
you need to rebase again your server files are different from one on develop

allebd commented on coverall.yml:
```diff
@@ -0,0 +1,2 @@
+service_name: travis-pro
+repo_token: UlARnqYpnmnGghoCE6IDSshcrqo1uPJ0z
```
New line required

allebd commented on coverals.yml:
```diff
@@ -0,0 +1,2 @@
+service_name: travis-pro
```
Your file is name coverals.yml instead of .coveralls.yml

allebd commented on package.json:
```diff
@@ -9,13 +9,12 @@
         "db:seed": "sequelize db:seed:all",
-        "coverage": "npm run generate-lcov"
-    },
+        "coverage": "nyc report --reporter=text-lcov | coveralls",
```
You have a comma at the end of this line which is not meant to be there

allebd commented on package.json:
```diff
@@ -9,13 +9,12 @@
-        "coverage": "npm run generate-lcov"
-    },
+        "coverage": "nyc report --reporter=text-lcov | coveralls",
+},
```
The closing bracket is wrongly positioned

Adekoreday replied:
the script to generate the lcov file will be used by coverage and codeclimate

henryade reviewed (commented):
repair your commit message and i will approve it
here is my commit message, please high light the wrong part.
ch(integrate)coveralls and codeclimate 
- coveralls
- add coveralls badge to readme.md file
- add codeclimate badge to readme.md file
- add coveralls-coveraget script
- add coverage script
[#166214085]

Lundii commented:
> repair your commit message and i will approve it

Nenny rewrite you commit to follow the convention.

Adekoreday reviewed (changes requested):
i think you need to handle code coverage reporting script as well 
i can only see coveralls
a better approach might to put the previous generate lcov script back and use the script below

kindly explain any other goals concerning this task

"coveralls-coverage": "coveralls < lcov.info",
    "codeclimate-coverage": "codeclimate-test-reporter < lcov.info",

for code climate and coveralls 
that script you inserted is only reporting to coveralls

you might install coveralls and codeclimate test reporter package

Nennyfills commented:
> i think you need to handle code coverage reporting script as well
> i can only see coveralls
> a better approach might to put the previous generate lcov script back and use the script below
> kindly explain any other goal behind this task

> "coveralls-coverage": "coveralls < lcov.info",
> "codeclimate-coverage": "codeclimate-test-reporter < lcov.info",
> 
> for code climate and coveralls
> that script you inserted is only reporting to coveralls

please, can you research on this?
> "coverage": "nyc report --reporter=text-lcov | coveralls" is also equal 
> "coveralls-coverage": "coveralls < lcov.info",
> "codeclimate-coverage": "codeclimate-test-reporter < lcov.info",

ok will do

Nennyfills commented:
> i think you need to handle code coverage reporting script as well
> i can only see coveralls
> a better approach might to put the previous generate lcov script back and use the script below
> 
> "coveralls-coverage": "coveralls < lcov.info",
> "codeclimate-coverage": "codeclimate-test-reporter < lcov.info",
> 
> for code climate and coveralls
> that script you inserted is only reporting to coveralls
> 
> you might install coveralls and codeclimate test reporter package
We are already using coveralls to report test coverage, please can you explain to me why we need code climate text coverage as well, meaning you want us to have two test coverage

Nennyfills commented:
> > repair your commit message and i will approve it
> 
> Nenny rewrite you commit to follow the convention.

here is my commit message, please high light the wrong part.
ch(integrate)coveralls and codeclimate
coveralls
add coveralls badge to readme.md file
add codeclimate badge to readme.md file
add coveralls-coveraget script
add coverage script
[#166214085]

ok you mean your PR 
I can relate with this ok
will proceed to test your work locally

Nennyfills committed:
166214085-story(integrate)coveralls and codeclimate
- install coveralls
- add coveralls badge to readme.md file
- add codeclimate badge to readme.md file
- add coverage script
[#166214085]

Lundii reviewed (approved):
Great
